### PR TITLE
Budget app

### PR DESCRIPTION
**Pull request by Sahar-SE**
### This PR Contains:

- Splash screen
- A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
- Sign up and log in pages
- The user should be able to register in the app with full name, email and password (all mandatory).
- The user can log into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
- Home page (catagories page)
- When the user logs in, they are presented with the categories page.
- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
- Transactions page
- For a given category, the list of transactions is presented, ordered by the most recent.
- At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new category" page
- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new transaction" page
- The user fills out a form to create a new transaction with:
- name (mandatory)
- amount (mandatory)
- categories (mandatory at least one)
- The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
-  Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

[Live Demo](https://budget-ruby-on-rails.herokuapp.com/)

Screenshot (121)

### CODE REVIEW

File: app/views/budgets/_budget_card.html.erb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+          <td><%= index + 1 %></td>
+          <td><%= budget.name %></td>
+          <td>$ <%= budget.amount %></td>
+          </td>
+        </tr>
+      <% end %>
+    </tbody>
+  </table>
+    </div>
+</div>
```
Comment (Deepakdanger) [inline app/views/budgets/_budget_card.html.erb:22]: - [x] I would suggest keeping the design same.

## Your design
Screenshot from 2022-10-22 02-09-41


## Given design
Screenshot from 2022-10-22 01-53-34

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @Sahar-SE, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 

## Required changes

- [x] The project should deploy the application using Heroku, Netlify or GitHub Pages.
- [x] The README should have steps to run the project locally.

```
$ git clone yourrepo.git
$ cd yourrepo
$ bundle
$ rails db:setup
$ rails s
```
_Check the comments under the review._


## Optional changes
You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/views/budgets/_budget_card.html.erb (hunk 1 above)
Comment (Sahar-SE) [reply app/views/budgets/_budget_card.html.erb:22]: **Thank You Sir I brought the Changes**

Screenshot (123)

**Review by descholar-ceo — APPROVED** [review]
Hi @Sahar-SE 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented all requested changes ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**